ql/src/java/org/apache/hadoop/hive/ql/parse/TezCompiler.java

```diff
@@ -1766,6 +1774,59 @@ private void removeSemijoinOptimizationByBenefit(OptimizeTezProcContext procCtx)
       GenTezUtils.removeBranch(rs);
       GenTezUtils.removeSemiJoinOperator(procCtx.parseContext, rs, ts);
     }
+
+    for (Entry<FilterOperator, Collection<SemijoinOperatorInfo>> e : globalReductionFactorMap.asMap().entrySet()) {
```

Comment:
Creating separate method for this and adding comments to explain why we are doing it will be nice.

Reply:
Good idea, will do.